File: backend/app/domain/models/offer.py
from datetime import datetime

from sqlalchemy import Column, Integer, String, Numeric, Boolean, DateTime
from sqlalchemy.orm import relationship

from app.domain.models.base import Base

class Offer(Base):
    __tablename__ = "Ofertas"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    nombre = Column(String(255), nullable=False)
    descripcion = Column(String, nullable=True)
    precio = Column(Numeric(10, 2), nullable=False)
    activo = Column(Boolean, nullable=False, default=True)
    fecha_creacion = Column(DateTime, default=datetime.now)
    fecha_actualizacion = Column(DateTime, default=datetime.now)

    productos = relationship(
        "OfferItem",
        back_populates="oferta",
        cascade="all, delete-orphan"
    )

    def __repr__(self):
        return f"<Offer(nombre={self.nombre}, precio={self.precio})>"

File: frontend/src/types/offer.ts
import type { OfferItem, OfferItemRequest } from './offer_item';

// Re-exportar OfferItem para facilitar imports
export type { OfferItem } from './offer_item';

export interface Offer {
  id: number;
  nombre: string;
  descripcion?: string | null;
  precio: number;
  activo: boolean;
  fecha_creacion: string;
  fecha_actualizacion: string;
  productos?: OfferItem[];
}

export interface CreateOfferRequest {
  nombre: string;
  descripcion?: string;
  precio: number;
  productos: OfferItemRequest[];
}

export interface UpdateOfferRequest {
  nombre?: string;
  descripcion?: string;
  precio?: number;
  productos?: OfferItemRequest[];
}
